messaging/src/main/java/org/axonframework/messaging/Message.java

```diff
@@ -75,9 +75,18 @@ public interface Message<T> extends Serializable {
      * @return the type of payload.
      * @deprecated Payloads are just jvm-internal representations. No need for matching against payload types
      */
-    @Deprecated // TODO #3085 - Introduce QualifiedName
+    @Deprecated // TODO #3085 - Replace for getMessageType once fully integrated
     Class<T> getPayloadType();
 
+    /**
+     * Returns the {@link QualifiedName message type} of this {@link Message}.
+     *
+     * @return The {@link QualifiedName message type} of this {@link Message}.
+     */
+    default QualifiedName getMessageType() {
```

Comment:
Given the method is defined on the Message interface, I suggest simply naming it `type()`

Reply:
Sounds fair to me, I'll rename it!